droidlet/lowlevel/hello_robot/remote/iphone_slam.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import threading
+
+import numpy as np
+from scipy.spatial.transform import Rotation as R
+import rospy
+from geometry_msgs.msg import Pose
+
+from iphone_reader import iPhoneReader
+
+log = logging.getLogger(__name__)
+
+IPHONE_OFFSET = [0, 0.1]  # XY offset in base frame
+IPHONE_ROT = [
+    [0.0, 1.0, 0.0],
+    [0.0, 0.0, -1.0],
+    [-1.0, 0.0, 0.0],
+]
+
+
+class Record3dSLAM:
+    def __init__(self):
+        self.reader = iPhoneReader(retrieve_imgs=False)
+
+        self.base_pose = np.zeros(3)
+        self.pose_lock = threading.Lock()
+
+        # Init publishers
+        self.c_pose_pub = rospy.Publisher("r3d_slam/camera_pose", Pose, queue_size=1)
```

Comment:
Consider using http://docs.ros.org/en/noetic/api/geometry_msgs/html/msg/PoseStamped.html
which timestamp from Record3d or generate time here can be applied